src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,34 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_USERS_AGE = 18;
+    private static final int MAX_USERS_AGE = 150;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUsersData(user);
+        return storageDao.add(user);
+    }
+
+    private void checkUsersData(User user) {
+        if (user == null) {
+            throw new RegistrationException("Input data is null.");
+        }
+        if (user.getAge() == null
+                || user.getAge() < MIN_USERS_AGE
+                || user.getAge() > MAX_USERS_AGE) {
+            throw new RegistrationException("User's age must be between "
+                    + MIN_USERS_AGE + " and " + MAX_USERS_AGE + ".");
+        }
+        if (user.getPassword() == null
+                || user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("Password length must be at least "
+                    + MIN_PASSWORD_LENGTH + " characters.");
+        }
+        if (user.getLogin() == null || storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login is already exists.");
+        }
```

Comment:
It would be easier to read your code if you create private methods for these checks:
```suggestion
        checkAge(user.getAge());
        checkPassword(user.getPassword());
        checkLogin(user.getLogin());       
```